clients/src/main/java/org/apache/kafka/common/config/AbstractConfig.java

```diff
@@ -355,6 +355,14 @@ public Map<String, Object> valuesWithPrefixAllOrNothing(String prefix) {
         return nonInternalConfigs;
     }
 
+    protected void clearConfig(String key) {
+        Object value = values.get(key);
+        if (value instanceof Number)
+            values.put(key, 0);
+        else
+            values.remove(key);
+    }
```

Comment:
I found that `session.timeout.ms` config should not be empty,  when create the consumer, I am not sure it's a bug or not.
when clearConfig method like, it would throw an error about 
```java
    protected void clearConfig(String key) {
            values.remove(key);
    }
```
```
Unknown configuration 'session.timeout.ms'
org.apache.kafka.common.config.ConfigException: Unknown configuration 'session.timeout.ms'
```
To avoid this case, I transfer the value to zero.

Reply:
> To avoid this case, I transfer the value to zero.

How to reproduce the error?

Reply:
1. modify the `clearConfig` method to 
   ```java
   protected void clearConfig(String key) {
        values.remove(key);
    }
   ```
2. run DescribeConsumerGroupTest, and the raft combined mode will error
![CleanShot 2024-09-09 at 08 41 31@2x](https://github.com/user-attachments/assets/41781964-2af0-4846-a803-e5c7232cc4ba)

Reply:
The main reason is in `GroupRebalanceConfig`, the `session.timeout.ms` config in necessary that no matter consumer protocol is, so we should keep it in config map, can't remove it from key set. https://github.com/apache/kafka/blob/f1e7954ad1dd2dc865195cdb796e2b1d518f8b85/clients/src/main/java/org/apache/kafka/clients/GroupRebalanceConfig.java#L50

Reply:
While I would love to change `GroupRebalanceConfig` (KAFKA-17823) to clean things up, would it work to update the `GroupRebalanceConfig` constructor for now?

```java
if (protocolType == ProtocolType.CONSUMER && config.getString(ConsumerConfig.GROUP_PROTOCOL_CONFIG).equals(GroupProtocol.CONSUMER.name())) {
    this.sessionTimeoutMs = 0;
} else {
    this.sessionTimeoutMs = config.getInt(CommonClientConfigs.SESSION_TIMEOUT_MS_CONFIG);
}
```

Reply:
this way can't resolve the problem about logging config which will misdirect user. see [comment](https://github.com/apache/kafka/pull/16899#discussion_r1743632489).
We should remove these misdirect config before print log, but print log is in the `AbstractConfig` contructor, so it is  little difficult to resolve this problem, we file an another [Jira](https://issues.apache.org/jira/browse/KAFKA-17821) to resolve this problem, We can focus on `ConsumerConfig` in this PR.

Reply:
It seens can resolve `logAll` which will misdirect problem, however I'm concerned it may not be the best solution. This approach would require `GroupRebalanceConfig` to be aware of both `ConsumerConfig` and `ProtocolType`, introducing unnecessary complexity as these config classes would depend on each other.

To avoid this, we have filed a new [JIRA ticket](https://issues.apache.org/jira/browse/KAFKA-17821) to focus specifically on this issue. For now, we can concentrate on improving ConsumerConfig in this PR.